configstack/running_module.go

```diff
@@ -184,13 +202,23 @@ func (module *runningModule) runNow() error {
 	}
 }
 
+var separator = strings.Repeat("-", 132)
+
 // Record that a module has finished executing and notify all of this module's dependencies
 func (module *runningModule) moduleFinished(moduleErr error) {
-	if moduleErr == nil {
-		module.Module.TerragruntOptions.Logger.Printf("Module %s has finished successfully!", module.Module.Path)
-	} else {
-		module.Module.TerragruntOptions.Logger.Printf("Module %s has finished with an error: %v", module.Module.Path, moduleErr)
+	status := "successfully!"
+	if moduleErr != nil {
+		status = fmt.Sprintf("with an error: %v", moduleErr)
+	}
+
+	module.Mutex.Lock()
+	defer module.Mutex.Unlock()
```

Comment:
Gah, Windows :(

The lock code is currently fairly simple, but typically, locks only get more and more complicated. Moreover, they don't compose. As a result, I'd like to think about an alternative strategy.

For example, one option is to have a goroutine listening for strings on a channel. Every string it gets, it logs to stdout. You could then send strings to that channel any time you want to write them to `stdout`, and you don't need any lock, as the channel takes care of ordering for you.